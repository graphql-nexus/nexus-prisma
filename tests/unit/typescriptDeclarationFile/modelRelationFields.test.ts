import { generateModules } from '../../__helpers__'

it('A model relation field projects a default resolve function', async () => {
  const { indexdts } = await generateModules(`
  model User {
    id    Int     @id
    posts Post[]
  }
  
  model Post {
    id        Int      @id
    author    User?    @relation(fields: [authorId], references: [id])
    authorId  Int?
  }
  `)

  expect(indexdts).toMatchSnapshot()
})
