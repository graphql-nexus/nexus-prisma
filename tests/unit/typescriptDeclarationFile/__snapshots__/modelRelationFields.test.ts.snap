// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A model relation field projects a default resolve function: index.d.ts 1`] = `
"import * as Nexus from 'nexus'
import * as NexusCore from 'nexus/dist/core'

//
//
// TYPES
// TYPES
// TYPES
// TYPES
//
//

declare namespace $Types {
  // Models

  /**
    * Nexus \`objectType\` related configuration based on the \`User\` model in your Prisma schema.
    *
    * ### ️⚠️ Missing Documentation for this Model
    * 
    * Automatically ✨ enrich this JSDoc with information about your enum
    * type by documenting it in your Prisma schema. For example:
    *
    * 
    * \`\`\`prisma
    * /// Lorem ipsum dolor sit amet...
    * model User {
    *   foo  String
    * }
    * \`\`\`
    * 
    * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
    *
    * @example
    *
    * import { objectType } from 'nexus'
    * import { User } from 'nexus-prisma'
    *
    * objectType({
    *   name: User.$name
    *   description: User.$description
    *   definition(t) {
    *     t.field(User.id.name, User.id)
    *   }
    * }) 
    */
  interface User {
    $name: 'User'
    $description: undefined
    /**
     * Nexus \`t.field\` related configuration based on the \`User.id\` field in your Prisma schema.
      *
      * ### ️⚠️ Missing Documentation for this Model
      * 
      * Automatically ✨ enrich this JSDoc with information about your enum
      * type by documenting it in your Prisma schema. For example:
      *
      * \`\`\`prisma
      * model User {
      *   /// Lorem ipsum dolor sit amet.
      *   id  Int
      * }
      * \`\`\`
      *
      * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
      *
      * @example
      *
      * import { objectType } from 'nexus'
      * import { User } from 'nexus-prisma'
      *
      * objectType({
      *   name: User.$name
      *   description: User.$description
      *   definition(t) {
      *     t.field(User.id.name, User.id)
      *   }
      * }) 
      */
    id: {
      /**
       * The name of this field.
       */
      name: 'id'
    
      /**
       * The type of this field.
       */
      type: NexusCore.NexusNonNullDef<'Int'>
    
      /**
       * The documentation of this field.
       */
      description: undefined
    
      /**
       * The resolver of this field
       */
      resolve: NexusCore.FieldResolver<'User', 'id'>
    }
    /**
     * Nexus \`t.field\` related configuration based on the \`User.posts\` field in your Prisma schema.
      *
      * ### ️⚠️ Missing Documentation for this Model
      * 
      * Automatically ✨ enrich this JSDoc with information about your enum
      * type by documenting it in your Prisma schema. For example:
      *
      * \`\`\`prisma
      * model User {
      *   /// Lorem ipsum dolor sit amet.
      *   posts  Post
      * }
      * \`\`\`
      *
      * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
      *
      * @example
      *
      * import { objectType } from 'nexus'
      * import { User } from 'nexus-prisma'
      *
      * objectType({
      *   name: User.$name
      *   description: User.$description
      *   definition(t) {
      *     t.field(User.posts.name, User.posts)
      *   }
      * }) 
      */
    posts: {
      /**
       * The name of this field.
       */
      name: 'posts'
    
      /**
       * The type of this field.
       */
      type: NexusCore.ListDef<Post> | NexusCore.NexusNonNullDef<Post>
    
      /**
       * The documentation of this field.
       */
      description: undefined
    
      /**
       * The resolver of this field
       */
      resolve: NexusCore.FieldResolver<'User', 'posts'>
    }
  }
  
  /**
    * Nexus \`objectType\` related configuration based on the \`Post\` model in your Prisma schema.
    *
    * ### ️⚠️ Missing Documentation for this Model
    * 
    * Automatically ✨ enrich this JSDoc with information about your enum
    * type by documenting it in your Prisma schema. For example:
    *
    * 
    * \`\`\`prisma
    * /// Lorem ipsum dolor sit amet...
    * model Post {
    *   foo  String
    * }
    * \`\`\`
    * 
    * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
    *
    * @example
    *
    * import { objectType } from 'nexus'
    * import { Post } from 'nexus-prisma'
    *
    * objectType({
    *   name: Post.$name
    *   description: Post.$description
    *   definition(t) {
    *     t.field(Post.id.name, Post.id)
    *   }
    * }) 
    */
  interface Post {
    $name: 'Post'
    $description: undefined
    /**
     * Nexus \`t.field\` related configuration based on the \`Post.id\` field in your Prisma schema.
      *
      * ### ️⚠️ Missing Documentation for this Model
      * 
      * Automatically ✨ enrich this JSDoc with information about your enum
      * type by documenting it in your Prisma schema. For example:
      *
      * \`\`\`prisma
      * model Post {
      *   /// Lorem ipsum dolor sit amet.
      *   id  Int
      * }
      * \`\`\`
      *
      * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
      *
      * @example
      *
      * import { objectType } from 'nexus'
      * import { Post } from 'nexus-prisma'
      *
      * objectType({
      *   name: Post.$name
      *   description: Post.$description
      *   definition(t) {
      *     t.field(Post.id.name, Post.id)
      *   }
      * }) 
      */
    id: {
      /**
       * The name of this field.
       */
      name: 'id'
    
      /**
       * The type of this field.
       */
      type: NexusCore.NexusNonNullDef<'Int'>
    
      /**
       * The documentation of this field.
       */
      description: undefined
    
      /**
       * The resolver of this field
       */
      resolve: NexusCore.FieldResolver<'Post', 'id'>
    }
    /**
     * Nexus \`t.field\` related configuration based on the \`Post.author\` field in your Prisma schema.
      *
      * ### ️⚠️ Missing Documentation for this Model
      * 
      * Automatically ✨ enrich this JSDoc with information about your enum
      * type by documenting it in your Prisma schema. For example:
      *
      * \`\`\`prisma
      * model Post {
      *   /// Lorem ipsum dolor sit amet.
      *   author  User?
      * }
      * \`\`\`
      *
      * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
      *
      * @example
      *
      * import { objectType } from 'nexus'
      * import { Post } from 'nexus-prisma'
      *
      * objectType({
      *   name: Post.$name
      *   description: Post.$description
      *   definition(t) {
      *     t.field(Post.author.name, Post.author)
      *   }
      * }) 
      */
    author: {
      /**
       * The name of this field.
       */
      name: 'author'
    
      /**
       * The type of this field.
       */
      type: NexusCore.NexusNullDef<'User'>
    
      /**
       * The documentation of this field.
       */
      description: undefined
    
      /**
       * The resolver of this field
       */
      resolve: NexusCore.FieldResolver<'Post', 'author'>
    }
    /**
     * Nexus \`t.field\` related configuration based on the \`Post.authorId\` field in your Prisma schema.
      *
      * ### ️⚠️ Missing Documentation for this Model
      * 
      * Automatically ✨ enrich this JSDoc with information about your enum
      * type by documenting it in your Prisma schema. For example:
      *
      * \`\`\`prisma
      * model Post {
      *   /// Lorem ipsum dolor sit amet.
      *   authorId  Int?
      * }
      * \`\`\`
      *
      * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
      *
      * @example
      *
      * import { objectType } from 'nexus'
      * import { Post } from 'nexus-prisma'
      *
      * objectType({
      *   name: Post.$name
      *   description: Post.$description
      *   definition(t) {
      *     t.field(Post.authorId.name, Post.authorId)
      *   }
      * }) 
      */
    authorId: {
      /**
       * The name of this field.
       */
      name: 'authorId'
    
      /**
       * The type of this field.
       */
      type: NexusCore.NexusNullDef<'Int'>
    
      /**
       * The documentation of this field.
       */
      description: undefined
    
      /**
       * The resolver of this field
       */
      resolve: NexusCore.FieldResolver<'Post', 'authorId'>
    }
  }

  // Enums

  // N/A –– You have not defined any models in your Prisma schema file.
}


//
//
// EXPORTS
// EXPORTS
// EXPORTS
// EXPORTS
//
//

//
//
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
//
//

/**
  * Nexus \`objectType\` related configuration based on the \`User\` model in your Prisma schema.
  *
  * ### ️⚠️ Missing Documentation for this Model
  * 
  * Automatically ✨ enrich this JSDoc with information about your enum
  * type by documenting it in your Prisma schema. For example:
  *
  * 
  * \`\`\`prisma
  * /// Lorem ipsum dolor sit amet...
  * model User {
  *   foo  String
  * }
  * \`\`\`
  * 
  * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
  *
  * @example
  *
  * import { objectType } from 'nexus'
  * import { User } from 'nexus-prisma'
  *
  * objectType({
  *   name: User.$name
  *   description: User.$description
  *   definition(t) {
  *     t.field(User.id.name, User.id)
  *   }
  * }) 
  */
export const User: $Types.User

/**
  * Nexus \`objectType\` related configuration based on the \`Post\` model in your Prisma schema.
  *
  * ### ️⚠️ Missing Documentation for this Model
  * 
  * Automatically ✨ enrich this JSDoc with information about your enum
  * type by documenting it in your Prisma schema. For example:
  *
  * 
  * \`\`\`prisma
  * /// Lorem ipsum dolor sit amet...
  * model Post {
  *   foo  String
  * }
  * \`\`\`
  * 
  * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
  *
  * @example
  *
  * import { objectType } from 'nexus'
  * import { Post } from 'nexus-prisma'
  *
  * objectType({
  *   name: Post.$name
  *   description: Post.$description
  *   definition(t) {
  *     t.field(Post.id.name, Post.id)
  *   }
  * }) 
  */
export const Post: $Types.Post

//
//
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
//
//

// N/A –– You have not defined any models in your Prisma schema file.

//
//
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
//
//

import { Runtime } from '../generator/runtime/settingsSingleton'

export const $settings: typeof Runtime.changeSettings
"
`;
